platform/src/main/java/org/hillview/storage/CassandraSSTableLoader.java

```diff
@@ -309,78 +270,77 @@ public ITable load(){
                     cds.add(cd);
                 }
                 assert this.actualSchema != null;
-                IColumnLoader loader = new CassandraSSTableLoader.SSTableColumnLoader(this.ssTableReader, this.actualSchema);
+                IColumnLoader loader = new CassandraSSTableLoader.SSTableColumnLoader(this.ssTableReader, 
+                    this.actualSchema);
                 return Table.createLazyTable(cds, size, this.filename, loader);
             } else {
-                return nonLazyLoading();
+                int sizeToLoad = this.actualSchema.getColumnCount();
+                // columns loader will load all column, so all item of columnToLoad need to be TRUE
+                boolean[] columnToLoad = new boolean[sizeToLoad];
+                Arrays.fill(columnToLoad, Boolean.TRUE);
+
+                ISSTableScanner currentScanner = this.ssTableReader.getScanner();
+                Spliterator<UnfilteredRowIterator> splititer = Spliterators.spliteratorUnknownSize(currentScanner, 
+                    Spliterator.IMMUTABLE);
+                Stream<UnfilteredRowIterator> partitions = StreamSupport.stream(splititer, false);
+
+                List<IColumn> arrColumns = this.loadColumns(columnToLoad, partitions, sizeToLoad);
+                return new Table(arrColumns, this.ssTablePath, null);
             }
         } catch (Exception ex) {
             throw new RuntimeException(ex);
         }
     }
-    
-    public ITable nonLazyLoading() {
-        assert this.actualSchema != null;
-        this.columns = this.actualSchema.createAppendableColumns();
 
-        /* Iterating each item inside the table */
-        ISSTableScanner currentScanner = this.ssTableReader.getScanner();
-        Spliterator<UnfilteredRowIterator> splititer = Spliterators.spliteratorUnknownSize(currentScanner, Spliterator.IMMUTABLE);
-        Stream<UnfilteredRowIterator> partitions = StreamSupport.stream(splititer, false);
+    /** Lazy and non-lazy loader will call this function to load specific column marked by columnToLoad */
+    private List<IColumn> loadColumns(boolean[] columnToLoad, Stream<UnfilteredRowIterator> partitions, int size) {
+        List<IAppendableColumn> columns = createColumns(columnToLoad, size);
+        // Iterating each item inside the table
         partitions.forEach(partition -> {
             int i;
             Object value;
-            Object[] line;
+            List<Object> line;
             Row row;
             Cell cell;
             AbstractType<?> cellType;
             Unfiltered unfiltered;
 
-            /* Serializing each partition */
-            while (partition.hasNext()){
+            // Serializing each partition
```

Comment:
this kind of comment does not add much, it is rather obvious what happens on the next line